File: ParcelableTest/app/src/main/java/com/example/parcelabletest/MainActivity.java
package com.example.parcelabletest;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.parcelabletest.bean.Book;

import java.io.InputStreamReader;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button parcelable = findViewById(R.id.parcelable);

        parcelable.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this,TestActivity.class);
                Book book = new Book();

                book.setName("第一行代码");
                book.setPrice("89");

                //传递对象
                intent.putExtra("data",book);
                startActivity(intent);
            }
        });
    }
}

File: SlideConflictDemo/app/src/main/java/com/example/slideconflictdemo/MainActivity.kt
package com.example.slideconflictdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    lateinit var mList: ArrayList<String>
    lateinit var mAddListView: ArrayList<View>
    lateinit var mBaseAdapter: BaseAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mAddListView = ArrayList()
        mList = ArrayList()
        for (i in 0..50) {
            mList.add(i, "data: $i")
        }
        for (i in 0..4){
            val listView = ListView(this)
            listView.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, mList)
            mAddListView.add(listView)
        }
        mBaseAdapter = BaseAdapter(mAddListView)
        baseViewPager.adapter = mBaseAdapter
    }
}